cpp/src/arrow/compute/exec/asof_join_node.cc

```diff
@@ -781,7 +781,8 @@ class AsofJoinNode : public ExecNode {
     // Prune memo entries that have expired (to bound memory consumption)
     if (!lhs.Empty()) {
       for (size_t i = 1; i < state_.size(); ++i) {
-        state_[i]->RemoveMemoEntriesWithLesserTime(lhs.GetLatestTime() - tolerance_);
+        OnType min_value = std::min<OnType>(0, lhs.GetLatestTime() - tolerance_);
```

Comment:
Do you mean
```suggestion
        OnType min_value = std::max<OnType>(0, lhs.GetLatestTime() - tolerance_);
```

Reply:
(I assume this is for when `lhs.GetLatestTime()` is smaller than the tolerance)

Reply:
Ah, yes.  I guess I got away with it because these are unsigned integers :laughing:

Reply:
Actually...I suppose I need to just do the if check for maximum readability, since the overflow will defeat `std::max`

```
OnType min_value = 0;
if (tolerance_ < lhs.GetLatestTime()) {
  min_value = lhs.GetLatestTime() - tolerance_;
}
```

Reply:
Actually, we don't need to prune if `min_value` is 0 so I was able to simplify it even further.